bn-apps/cordapp-updates-distribution/corda-updates-core/src/main/kotlin/net/corda/businessnetworks/cordaupdates/core/CordappSyncer.kt

```diff
@@ -0,0 +1,122 @@
+        val syncTask = syncerConf.findSourceFor(coordinatesWithRange) ?: throw CordappSourceNotFoundException(coordinatesWithRange)
+        val resolver = CordaMavenResolver.create(syncerConf, syncTask)
+        return listOf(resolver.resolveVersionRange(coordinatesWithRange, additionalConfigurationProperties))
+    }
+}
+
+/**
+ * [CordappSource] allows, that allows to associate multiple CorDapps with a remote repository
+ */
+data class CordappSource(
+        val remoteRepoUrl : String,
+        val cordapps : List<String>,
+        val httpUsername : String? = null,
+        val httpPassword : String? = null)
+
+/**
+ * Configuration for [CordappSyncer]. Usually is red from settings.conf
+ */
+data class SyncerConfiguration(
+        val localRepoPath : String,
+        val httpProxyHost : String? = null,
+        val httpProxyType : String? = null,
+        val httpProxyPort : Int? = null,
+        val httpProxyUsername : String? = null,
+        val httpProxyPassword : String? = null,
+        val rpcHost : String? = null,
+        val rpcPort : Int? = null,
+        val rpcUsername : String? = null,
+        val rpcPassword : String? = null,
+        val cordappSources : List<CordappSource>) {
+    companion object {
+        fun readFromFile(file : File) : SyncerConfiguration {
+
+            val conf = ConfigFactory.parseFile(file)!!
+            val cordapps = conf.getObjectList("cordappSources").asSequence().map { it.toConfig()!! }.map { CordappSource(
+                    it.getString("remoteRepoUrl"),
+                    it.getStringList("cordapps"),
+                    it.getStringOrNull("httpUsername"),
+                    it.getStringOrNull("httpPassword")
+            ) }.toList()
+
+            return SyncerConfiguration(
+                    conf.getString("localRepoPath"),
+                    conf.getStringOrNull("httpProxyHost"),
+                    conf.getStringOrNull("httpProxyType"),
+                    conf.getIntOrNull("httpProxyPort"),
+                    conf.getStringOrNull("httpProxyUsername"),
+                    conf.getStringOrNull("httpProxyPassword"),
+                    conf.getStringOrNull("rpcHost"),
+                    conf.getIntOrNull("rpcPort"),
+                    conf.getStringOrNull("rpcUsername"),
+                    conf.getStringOrNull("rpcPassword"),
+                    cordapps
+            )
+        }
+
+        private fun Config.getStringOrNull(key : String) = if (hasPath(key)) getString(key) else null
+        private fun Config.getIntOrNull(key : String) = if (hasPath(key)) getInt(key) else null
+    }
+
+    fun findSourceFor(cordapp : String) : CordappSource? {
+        val artifact = DefaultArtifact(cordapp)
+        return cordappSources.firstOrNull { it.cordapps.contains("${artifact.groupId}:${artifact.artifactId}") }
+    }
+
+}
+
+open class SyncerException(message : String? = null, cause : Throwable? = null) : Exception(message, cause)
```

Comment:
Needs KDocs, maybe make it sealed.

Reply:
Removed this exception as it hasn't been used really.